packages/amplify-cli/src/amplify-exception-handler.ts

```diff
@@ -47,25 +47,52 @@ export const handleException = async (exception: unknown): Promise<void> => {
     printAmplifyException(amplifyException);
   }
 
+  // Swallow and continue if any operations fail
   if (context) {
-    await reportError(context, amplifyException);
+    await executeSafely(reportError, [context, amplifyException], 'Failed to report error');
   }
 
-  await executeHooks(
-    HooksMeta.getInstance(undefined, 'post', {
+  await executeSafely(
+    executeHooks,
+    [HooksMeta.getInstance(undefined, 'post', {
       message: amplifyException.message ?? 'undefined error in Amplify process',
       stack: amplifyException.stack ?? 'undefined error stack',
-    }),
+    })],
+    'Failed to execute hooks',
   );
 
-  logger.logError({
-    message: amplifyException.message,
-    error: amplifyException,
-  });
+  await executeSafely(
+    logger.logError,
+    [{
+      message: amplifyException.message,
+      error: amplifyException,
+    }],
+    'Failed to log error',
+  );
 
   process.exitCode = 1;
 };
 
+/**
+ * Utility function to ensure a passed in function does not invoke the exception handler to avoid an infinite loop
+ *
+ * @param functionToExecute - the function that should be executed, but never reject
+ * @param parameters - parameters to pass to the function
+ * @param errorMessagePrefix - error message prefix before the thrown error is printed
+ *
+ * The type "Function" is banned by the linter, but I can't find a better way to define a more accurate type that
+ * allows an arbitrary number of parameters
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+const executeSafely = async (functionToExecute: Function, parameters: unknown[], errorMessagePrefix: string): Promise<void> => {
```

Comment:
nit: this could be made a little cleaner by just taking in a no-arg function:
```ts
const executeSafely = async (func: () => Promise<void>, errorPrefix: string): Promise<void> => {
  try {
    await func()
  } catch {
    // stuff
  }
}
```
then the usage would be:
```ts
await executeSafely(() => logger.logError({message, error}))
```

Reply:
I like this a lot better, I'll change it